clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/view/EcsApplicationProvider.java

```diff
@@ -0,0 +1,147 @@
+package com.netflix.spinnaker.clouddriver.ecs.view;
+
+
+import com.amazonaws.services.ecs.AmazonECS;
+import com.amazonaws.services.ecs.model.*;
+import com.google.common.collect.Sets;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonCredentials;
+import com.netflix.spinnaker.clouddriver.ecs.model.EcsApplication;
+import com.netflix.spinnaker.clouddriver.model.Application;
+import com.netflix.spinnaker.clouddriver.model.ApplicationProvider;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentials;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+@Component
+public class EcsApplicationProvider implements ApplicationProvider {
+
+  public static final String SERVICE_DELIMITER = "-";
+
+  AccountCredentialsProvider accountCredentialsProvider;
+  AmazonClientProvider amazonClientProvider;
+
+  @Autowired
+  public EcsApplicationProvider(AccountCredentialsProvider accountCredentialsProvider, AmazonClientProvider amazonClientProvider) {
+    this.accountCredentialsProvider = accountCredentialsProvider;
+    this.amazonClientProvider = amazonClientProvider;
+    getApplications(false);
+  }
+
+
+  @Override
+  public Application getApplication(String name) {
+
+    for (Application application: getApplications(false)) {
+      if (name.equals(application.getName())) {
+        return application;
```

Comment:
Style questions. In general I try to avoid returning in the middle of a function as the standard case, and only do these mid-function returns when it greatly improves readability (and try to minimize them). Would this not be better as:
```
Application result = null
for (...) {
  if (...) {
    result = application;
    break;
  }
}

return result;
```

I do not feel very strongly about a functions of this size, but as functions get bigger it becomes harder to read when there are multiple returns.

Reply:
In Java, the recommended approach is to actually return instead of using the approach you described
See for example https://stackoverflow.com/questions/4805258/is-it-bad-style-to-use-return-to-break-a-for-loop-in-java

It is also a recommendation in https://www.amazon.com/Clean-Code-Handbook-Software-Craftsmanship/dp/0132350882

Reply:
a more thorough SO discussion https://stackoverflow.com/questions/36707/should-a-function-have-only-one-return-statement 
And the quote for Clean Code that I could find, not exactly a recomendation but an accepted exception:

> "Some programmers follow Edsger Dijkstra‘s rules of structured programming. Dijkstra said that every function, and every block within a function, should have one entry and one exit. Following these rules means there should only be one return statement in a function, no break or continue statements in a loop…
> While we are sympathetic to the goals and disciplines of structured programming, those rules serve little benefit when functions are very small. It is only in larger functions that such rules provide significant benefit. So, if you keep your functions small, then the occasional multiple return, break, or continue statement does no harm and can sometimes even be more expressive then the single-entry, single-exit rule."

But as mentioned above, for a functions this small, having the multiple returns does no harm